conformance/tests/gateway-static-addresses.go

```diff
@@ -134,12 +134,27 @@ var GatewayStaticAddresses = suite.ConformanceTest{
 		})
 		kubernetes.GatewayStatusMustHaveListeners(t, s.Client, s.TimeoutConfig, gwNN, finalExpectedListenerState)
 		require.Len(t, currentGW.Spec.Addresses, 1, "expected only 1 address left specified on Gateway")
-		require.Len(t, currentGW.Status.Addresses, 1, "one usable address was provided, so it should be the one reflected in status")
-		require.Equal(t, usableAddress.Type, currentGW.Status.Addresses[0].Type, "expected address type to match the usable address")
-		require.Equal(t, usableAddress.Value, currentGW.Status.Addresses[0].Value, "expected usable address to be assigned")
+		require.NotContains(t, extractStatusAddresses(currentGW.Status.Addresses), unusableAddress.Value, "should not contain the unusable address")
+		require.NotContains(t, extractStatusAddresses(currentGW.Status.Addresses), invalidAddress.Value, "should not contain the invalid address")
+		require.Contains(t, extractStatusAddresses(currentGW.Status.Addresses), usableAddress.Value, "should contain the usable address")
```

Comment:
Nit: Would be a bit cleaner to run `extractStatusAddresses` once instead of three times.

Reply:
+1